sources/packages/backend/apps/api/src/route-controllers/program-year/program-year.institutions.controller.ts

```diff
@@ -0,0 +1,38 @@
+import { Controller, Get, NotFoundException } from "@nestjs/common";
+import { ProgramYearService } from "../../services";
+import { ClientTypeBaseRoute } from "../../types";
+import BaseController from "../BaseController";
+import { AllowAuthorizedParty } from "../../auth/decorators/authorized-party.decorator";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { OptionItemAPIOutDTO } from "../models/common.dto";
+import { IsBCPublicInstitution } from "../../auth/decorators";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("program-year")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-program-year`)
+export class ProgramYearInstitutionsController extends BaseController {
+  constructor(private readonly programYearService: ProgramYearService) {
+    super();
+  }
+
+  /**
+   * Gets a list of program years returned as option items (id/description pair).
+   * @returns an array of program years as id/description objects.
+   */
+  @IsBCPublicInstitution()
```

Comment:
It is unlikely that the non-public institutions will have access to the method under this controller. I would recommend moving the `@IsBCPublicInstitution` to the class level.